source/pages/Settings/SwitchEthereumChain.tsx

```diff
@@ -24,7 +29,7 @@ const SwitchChain: React.FC = () => {
   const onSubmit = async () => {
     setLoading(true);
     try {
-      wallet.setActiveNetwork(network, 'ethereum');
+      await wallet.setActiveNetwork(network, 'ethereum');
```

Comment:
This might make a popup to be open for a long time, what happens if the user manually closes the popUp whilst the chain switch is happening ?

Reply:
The average popup opening time is 5 seconds, it's quite acceptable, no? If user manually closes, the network will change normally, but in the background

Reply:
Yup, I tested further today and got no time higher than 5 seconds. Tks man